Part 1. Техническая часть

1.1 Загрузка данных

In [251]:
import os
import glob
import pandas as pd

In [252]:
files = glob.glob('*.csv')
files

['reviews_argentina.csv',
 'reviews_armenia.csv',
 'reviews_australia.csv',
 'reviews_austria.csv',
 'reviews_bosnia_and_herzegovina.csv',
 'reviews_brazil.csv',
 'reviews_bulgaria.csv',
 'reviews_canada.csv',
 'reviews_chile.csv',
 'reviews_china.csv',
 'reviews_croatia.csv',
 'reviews_cyprus.csv',
 'reviews_czech_republic.csv',
 'reviews_egypt.csv',
 'reviews_england.csv',
 'reviews_france.csv',
 'reviews_georgia.csv',
 'reviews_germany.csv',
 'reviews_greece.csv',
 'reviews_hungary.csv',
 'reviews_india.csv',
 'reviews_israel.csv',
 'reviews_italy.csv',
 'reviews_lebanon.csv',
 'reviews_luxembourg.csv',
 'reviews_macedonia.csv',
 'reviews_mexico.csv',
 'reviews_moldova.csv',
 'reviews_morocco.csv',
 'reviews_new_zealand.csv',
 'reviews_peru.csv',
 'reviews_portugal.csv',
 'reviews_romania.csv',
 'reviews_serbia.csv',
 'reviews_slovakia.csv',
 'reviews_slovenia.csv',
 'reviews_south_africa.csv',
 'reviews_spain.csv',
 'reviews_switzerland.csv',
 'reviews_turkey.csv',
 'reviews_ukrain

In [253]:
combined = pd.DataFrame()

for file in files:
    data = pd.read_csv(file, sep=';')
    combined = pd.concat([combined, data])

In [254]:
combined

entry_id    country                                        description  \
0        119966  Argentina  Raspberry and cassis aromas are fresh and upri...   
1        119956  Argentina  Compared to the regular 2006 Malbec from Chaka...   
2        119950  Argentina  Andeluna's top wines tend to be ripe and plump...   
3        119943  Argentina  A lively, well-made blend of Tempranillo, Malb...   
4        119939  Argentina  There is a select group of under-$20 Malbecs f...   
...         ...        ...                                                ...   
63063        13         US  Slightly reduced, this wine offers a chalky, t...   
63064        11         US  Soft, supple plum envelopes an oaky structure ...   
63065         5         US  Much like the regular bottling from 2012, this...   
63066         4         US  Pineapple rind, lemon pith and orange blossom ...   
63067         3         US  Tart and snappy, the flavors of lime flesh and...   

                              designation  points  price          province  \
0                                Pedernal      90   43.0             Other   
1                                 Reserve      91   25.0  Mendoza Province   
2                               Pasionado      91   55.0  Mendoza Province   
3                                  B Crux      91   24.0  Mendoza Province   
4                      La Madras Vineyard      91   18.0  Mendoza Province   
...                                   ...     ...    ...               ...   
63063                                 NaN      87   34.0        California   
63064                      Mountain Cuvée      87   19.0        California   
63065  Vintner's Reserve Wild Child Block      87   65.0            Oregon   
63066                Reserve Late Harvest      87   13.0          Michigan   
63067                                 NaN      87   14.0            Oregon   

                    region         taster_name  \
0                 San Juan   Michael Schachner   
1            Luján de Cuyo   Michаеl Schаchnеr   
2               Uco Valley   Michael Schachner   
3               Uco Valley   Michаel Schаchner   
4                  Mendoza   Michаеl Schаchnеr   
...                    ...                 ...   
63063     Alexander Valley      Virginiе Bооnе   
63064          Napa Valley      Virginiе Bооnе   
63065    Willamette Valley        Paul Gregutt   
63066  Lake Michigan Shore  Alexаnder Peаrtree   
63067    Willamette Valley        Paul Gregutt   

                                                   title             variety  \
0        Finca Las Moras 2010 Pedernal Malbec (San Juan)              Malbec   
1            Chakana 2006 Reserve Malbec (Luján de Cuyo)              Malbec   
2               Andeluna 2004 Pasionado Red (Uco Valley)           Red Blend   
3               O. Fournier 2005 B Crux Red (Uco Valley)           Red Blend   
4      Ricardo Santos 2006 La Madras Vineyard Malbec ...              Malbec   
...                                                  ...                 ...   
63063  Louis M. Martini 2012 Cabernet Sauvignon (Alex...  Cabernet Sauvignon   
63064  Kirkland Signature 2011 Mountain Cuvée Caberne...  Cabernet Sauvignon   
63065  Sweet Cheeks 2012 Vintner's Reserve Wild Child...          Pinot Noir   
63066  St. Julian 2013 Reserve Late Harvest Riesling ...            Riesling   
63067      Rainstorm 2013 Pinot Gris (Willamette Valley)          Pinot Gris   

                   winery  
0         Finca Las Moras  
1                 Chakana  
2                Andeluna  
3             O. Fournier  
4          Ricardo Santos  
...                   ...  
63063    Louis M. Martini  
63064  Kirkland Signature  
63065        Sweet Cheeks  
63066          St. Julian  
63067           Rainstorm  

[149908 rows x 12 columns]

In [255]:
combined.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 149908 entries, 0 to 63067
Data columns (total 12 columns):
 #   Column       Non-Null Count   Dtype  
---  ------       --------------   -----  
 0   entry_id     149908 non-null  int64  
 1   country      149908 non-null  object 
 2   description  149908 non-null  object 
 3   designation  106749 non-null  object 
 4   points       149908 non-null  int64  
 5   price        139418 non-null  float64
 6   province     149908 non-null  object 
 7   region       125516 non-null  object 
 8   taster_name  118829 non-null  object 
 9   title        149908 non-null  object 
 10  variety      149906 non-null  object 
 11  winery       149908 non-null  object 
dtypes: float64(1), int64(2), object(9)
memory usage: 14.9+ MB


Вопросы:
1. Какое количество строк в загруженном dataframe? 
Ответ: 149908
2. Какое количество столбцов в загруженном dataframe?
Ответ: 12

1.2 Очистка данных

Удаляем дупликаты в датафрейме

In [256]:
# посмотрим сколько дупликатов

duplicated = combined[combined.duplicated(subset=['entry_id'])]
len(duplicated)

29979

In [257]:
# удалим дупликаты

no_dupl_df = combined.drop_duplicates(subset=['entry_id'])
len(no_dupl_df)

119929

Чистим от кирилических букв

In [258]:
taster_name = no_dupl_df['taster_name']
taster_name

0         Michael Schachner
1         Michаеl Schаchnеr
2         Michael Schachner
3         Michаel Schаchner
4         Michаеl Schаchnеr
                ...        
63062        Virginiе Boonе
63064        Virginiе Bооnе
63065          Paul Gregutt
63066    Alexаnder Peаrtree
63067          Paul Gregutt
Name: taster_name, Length: 119929, dtype: object

In [259]:
from transliterate import translit

transl_name = str(taster_name)
type(transl_name)

str

In [260]:
# на всякий случай сделаем проверку на кириллические символы

cyril = ('абвгдеёжзийклмнопрстуфхцчшщъыьэюя')

find_cyril = [x for x in cyril if x in transl_name]
find_cyril

['а', 'е', 'о']

In [261]:
# способ 1 - транлитерация всего текста, если мы не знаем, какие буквы на кирилице

transl_name = translit(transl_name, language_code='ru', reversed=True)
transl_name

'0         Michael Schachner\n1         Michael Schachner\n2         Michael Schachner\n3         Michael Schachner\n4         Michael Schachner\n                ...        \n63062        Virginie Boone\n63064        Virginie Boone\n63065          Paul Gregutt\n63066    Alexander Peartree\n63067          Paul Gregutt\nName: taster_name, Length: 119929, dtype: object'

In [262]:
# проверяем, осталась ли кирилица

cyril = ('абвгдеёжзийклмнопрстуфхцчшщъыьэюя')

find_cyril = [x for x in cyril if x in transl_name]
find_cyril

[]

In [263]:
# способоб 2 : так как нам известно, что на кирилице только "a", "o", "e", то можно заменить конкретно их

taster_name_new = no_dupl_df['taster_name'].str.lower()

trans = {'а':'a', 'o':'о', 'e':'е'}
for s in trans:
    taster_name_new = taster_name_new.replace(s, trans[s])

no_dupl_df['new_taster_name'] = taster_name_new.str.title()
no_dupl_df['new_taster_name']

C:\Users\vika1\AppData\Local\Temp\ipykernel_8580\150102333.py:9: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  no_dupl_df['new_taster_name'] = taster_name_new.str.title()


0         Michael Schachner
1         Michаеl Schаchnеr
2         Michael Schachner
3         Michаel Schаchner
4         Michаеl Schаchnеr
                ...        
63062        Virginiе Boonе
63064        Virginiе Bооnе
63065          Paul Gregutt
63066    Alexаnder Peаrtree
63067          Paul Gregutt
Name: new_taster_name, Length: 119929, dtype: object

In [264]:
# проверим второй способ

cyril = ('абвгдеёжзийклмнопрстуфхцчшщъыьэюя')

find_cyril = [x for x in cyril if x in taster_name_new]
find_cyril

[]

In [280]:
# посмотрим, сколько уникальных значений

no_dupl_df.taster_name.nunique()

74

Вопросы:
1. Какое количество строк в очищенном dataframe?
Ответ: 119929
2. Какое количество уникальных значений содержится в taster_name (вкл. NA )
Ответ: 74

1.3 Эксплораторный анализ

Задания:
1. Для переменных points и price приведите набор описательных статистик:
Среднее
Медиана
Стандартное отклонение

In [274]:
no_dupl_df['points'].mean()

88.44214493575366

In [272]:
no_dupl_df['points'].median()

88.0

In [275]:
no_dupl_df['points'].std()

3.0932746182684188

In [276]:
no_dupl_df['price'].mean()

35.6261094873496

In [277]:
no_dupl_df['price'].median()

25.0

In [278]:
no_dupl_df['price'].std()

42.11291028862946

2. Опишите связь между points и price
Укажите коэффициент корреляции.

In [279]:
# коэффициент корреляции points and price

no_dupl_df['points']. corr(no_dupl_df['price'])

0.4165616206391788